app/controllers/insights_cloud/hits_controller.rb

```diff
@@ -1,17 +1,39 @@
 module InsightsCloud
   class HitsController < ::ApplicationController
+    include Foreman::Controller::AutoCompleteSearch
+
     def index
+      page, per_page, search = insights_search_params.values_at(:page, :per_page, :search)
+      hits = InsightsHit.search_for(search)
```

Comment:
You can use [`resource_base_search_and_page`](https://github.com/theforeman/foreman/blob/7a4899774a957bf7336db12f5643ed21024a929c/app/controllers/application_controller.rb#L240) here.